### PR TITLE
Hardcoded maximum tls version to 1.3

### LINKED ISSUE #672: Automatically set tls_maximum_protocol_version to opt in to TLS 1.3
Please add support for the `tls_maximum_protocol_version` envoy api parameter (like you do for `tls_minimum_protocol_version`) to allow users to opt in to TLS 1.3. See https://github.com/envoyproxy/envoy/issues/3544.

### PR DESCRIPTION
**Pull request by cromefire**
Fixes #672

### CODE REVIEW

**Comment by cromefire** [comment]
I give up. Either the sign off thing or go is breaking it

**Comment by rosskukulinski** [comment]
Signoff checker issues resolved - sorry for the problems @cromefire

**Comment by cromefire** [comment]
@rosskukulinski in this PR sign off worked, but godep modified a file

**Comment by cromefire** [comment]
See #752 and #746

File: Gopkg.lock [hunk 1]
```diff
@@ -2,777 +2,320 @@
 
 
 [[projects]]
-  digest = "1:1acadbbc24182315b628f727b2e9ac653266d1644ca4007e0766c28110afc072"
```
Comment (cromefire) [inline Gopkg.lock]: Here is the problem and I don't want to redo this again (and my IDE does not want to let me revert this)

**Comment by cromefire** [comment]
okay I finally got it the revert got stuck somewhere

**Comment by rosskukulinski** [comment]
> could you please make checker look for gpg and author? that would make it possible to commit using GitHub or automatic IDE features.

I've submitted a request to our SRE team to take a look at the feasibility of that.

**Comment by cromefire** [comment]
Thanks

**Comment by cromefire** [comment]
Oh and look at the email address not the username, those are subject to change and/or be replaced with the display name.

**Comment by cromefire** [comment]
> I've submitted a request to our SRE team to take a look at the feasibility of that.

@rosskukulinski I played around with the github api and this is how you can get the signing:
```graphql
{
  repository(owner: "heptio", name: "contour") {
    pullRequest(number: 753) {  # PR number here
      commits(first: 100) {  # Implement some paging
        nodes {
          commit {
            signature {
              email
              isValid
              signer {
                login
                name
              }
            }
          }
        }
      }
    }
  }
}
```
Result:
```json
{
  "data": {
    "repository": {
      "pullRequest": {
        "commits": {
          "nodes": [
            {
              "commit": {
                "signature": {
                  "email": "tim.l@nghorst.net",
                  "isValid": true,
                  "signer": {
                    "login": "cromefire",
                    "name": "Cromefire_"
                  }
                }
              }
            },
            {
              "commit": {
                "signature": {
                  "email": "tim.l@nghorst.net",
                  "isValid": true,
                  "signer": {
                    "login": "cromefire",
                    "name": "Cromefire_"
                  }
                }
              }
            }
          ]
        }
      }
    }
  }
}
```
You can try it here: https://developer.github.com/v4/explorer/

Hope that helps

**Comment by davecheney** [comment]
Thank you for your persistence.

**Pull request merged**